### DISCUSSION TITLE
Function Does Not Exist With Plugin

### DISCUSSION THREAD

**Discussion by JRedOW** [prisma/prisma#10377] [Q&A] [answered]
I am trying to use TimeScaleDB.  The sql passes the shadow database and the main database when I run it myself, but throws a weird error when I migrate.

The Migration:
```
-- CreateTable
CREATE TABLE "song_time" (
    "userID" UUID NOT NULL,
    "modID" TEXT NOT NULL,
    "songID" INTEGER NOT NULL,
    "diffID" INTEGER NOT NULL,
    "time" TIMESTAMP(3) NOT NULL DEFAULT CURRENT_TIMESTAMP,
    "score" INTEGER NOT NULL
);

-- Verify Extention
CREATE EXTENSION IF NOT EXISTS timescaledb CASCADE;

-- ConvertHypertable
SELECT create_hypertable('song_time'::regclass, 'time'::name);
```

The Error:
```
Error: P3018

A migration failed to apply. New migrations cannot be applied before the error is recovered from. Read more about how to resolve migration issues in a production database: https://pris.ly/d/migrate-resolve

Migration name: 20211121045719_score_init

Database error code: 42883

Database error:
ERROR: function create_hypertable(regclass, name) does not exist
HINT: No function matches the given name and argument types. You might need to add explicit type casts.

Position:
 10
 11 -- Verify Extention
 12 CREATE EXTENSION IF NOT EXISTS timescaledb CASCADE;
 13
 14 -- ConvertHypertable
 15 SELECT create_hypertable('song_time'::regclass, 'time'::name);

DbError { severity: "ERROR", parsed_severity: Some(Error), code: SqlState("42883"), message: "function create_hypertable(regclass, name) does not exist", detail: None, hint: Some("No function matches the given name and argument types. You might need to add explicit type casts."), position: Some(Original(354)), where_: None, schema: None, table: None, column: None, datatype: None, constraint: None, file: Some("parse_func.c"), line: Some(624), routine: Some("ParseFuncOrColumn") }
```

Does anyone see something I don't?  Thanks!

**Comment by janpio** [comment]
Is is possible the main dev database indeed does not have the `timescaledb` extension enabled and enabling it in the migration might not work (e.g. inside the same transaction context)?

**Comment by JRedOW** [comment, marked as answer]
That's the strange part not only do I already enable it in the migration, but running that migration through raw requests outside of prisma works fine.  I've just given up.

**Reply by janpio** [reply]
Could you maybe quickly create a bug report issue with the information you already posted? I think this is worth reproducing on our side to figure out what might be going on.

**Reply by JRedOW** [reply]
#10388

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
